companion.py

```diff
@@ -355,33 +393,39 @@ def query(self, endpoint):
             self.retrying = False
             self.login()
             raise CredentialsError()
+
         elif 500 <= r.status_code < 600:
             # Server error. Typically 500 "Internal Server Error" if server is down
             self.dump(r)
             raise ServerError()
 
         try:
-            r.raise_for_status()	# Typically 403 "Forbidden" on token expiry
-            data = r.json()		# May also fail here if token expired since response is empty
+            r.raise_for_status()  # Typically 403 "Forbidden" on token expiry
+            data = r.json()  # May also fail here if token expired since response is empty
```

Comment:
These two are the lines that can throw exceptions, the below except clause should move to catching the HTTP exceptions (or just checking them instead of using raise_for_status, and the r.json can throw only so many errors